packages/core/admin/admin/src/content-manager/history/pages/History.tsx

```diff
@@ -100,33 +100,31 @@ const HistoryPage = () => {
     }
   }, [versionsResponse.isLoading, navigate, query.id, versionsResponse.data?.data, query]);
 
-  const { permissions, isLoading: isLoadingPermissions } = useSyncRbac(slug!, query, 'History');
-
-  if (isLoadingDocument || isLoadingLayout || versionsResponse.isLoading || isLoadingPermissions) {
-    return <Page.Loading />;
-  }
-
   /**
    * Ensure that we have the necessary data to render the page:
    * - slug for single types
    * - slug _and_ documentId for collection types
    */
-  if (!slug || (!documentId && collectionType === COLLECTION_TYPES)) {
+  if (!slug || (collectionType === COLLECTION_TYPES && !documentId)) {
     return <Navigate to="/content-manager" />;
   }
 
-  // TODO: real error state when designs are ready
-  if (versionsResponse.isError || !versionsResponse.data || !layout || !schema || !permissions) {
-    return null;
+  if (isLoadingDocument || isLoadingLayout || versionsResponse.isLoading || !selectedVersionId) {
+    return <Page.Loading />;
+  }
+
+  if (
+    !versionsResponse.isError &&
+    (!versionsResponse.data?.data || versionsResponse.data.data.length === 0)
+  ) {
+    return <Page.NoData />;
   }
 
-  const selectedVersion = versionsResponse.data.data.find(
+  const selectedVersion = versionsResponse.data?.data.find(
```

Comment:
isn't the second data the one that needs optional chaining? here if typescript is happy with this it means we can probably remove the ?

Reply:
I think TypeScript believes that either we will have a data key or an error key. As long we haven't handled the error case it complains here.